### DISCUSSION TITLE
Chromatic keyboard layout

### DISCUSSION THREAD

**Discussion by maks** [bangcorrupt/hacktribe#168] [Ideas]
Speaking of Loopop hacks from my last thread, I was reminded that he had [a "hack" using external midi to create a Chromatic keyboard](https://www.youtube.com/watch?v=mtO8_NJkK2I) layout like a traditional keyboard that makes way more sense (eg. the way Elektron machines do).

It would be **really** nice to have this implemented within Hacktribe rather than needing to have a external midi control app plugged in.

**Comment by bangcorrupt** [comment]
If I get as far as making hacktribe hackable there will be a template function where control input can be remapped.

**Reply by maks** [reply]
Sorry I should have explained this better. I wasn't really wanting to be able to do arbitrary controls remapping, rather just to change the Chromatic scale pad mapping permanently in Hacktribe. I guess it would be nice to be able to switch between the current layout and a standard keyboard one, but I suspect given the option of having one or the other, pretty much everyone would prefer a standard keyboard layout.
Would just permanently changing the layout be an easier proposition?

**Reply by bangcorrupt** [reply]
I don't like the keyboard layout hack, the keys aren't offset correctly and some pads are wasted, but I can't play piano anyway.  I'd like to be able to change the interval between rows, like tuning strings.  This could be integrated into the scale mode hacks.

Arbitrary control remapping is a stepping stone on the way to a lot of UI hacks, like a separate page for onboard FX editing.  If I can do it in an intuitive way that's decoupled from the factory firmware, people can set things up how they like.

**Reply by maks** [reply]
Haha I can't play piano either 😄, though just from my point of view as a newcomer learning music, so many synth resources are piano keyboard based that I do get quite lost trying to play notes in keyboard mode on the E2 pads. Watching videos of people using Elektron devices, with the same 2x8 pads layout but that has a keyboard-style chromatic layout, it does make more sense, at least to me.

I was just hoping this might only be a small tweak to make in the firmware...

But yes if definitely would be very nice to have user control remapping, as you say that would open up alot of possibilities.

**Reply by BKLronin** [reply]
I wished for a chromatic keyboard version too for a long time. Pre progammed scales are nice to get to know the character to each oft hem without the hassel of having to learn them. On th electribe however it would be a geat help to play along with other instruments or finally play freely. The pads and the leds could nicely light up with the not used buttons as sustain pedal or something similar useful.

I thought that this would be just another scale without the led lights but it seems its more complicated.
